The job **`x86_64-gnu-llvm-12`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/7550306979?check_suite_focus=true) [(plain)](https://github.com/rust-lang/rust/commit/94663888e3f2cfe818493c7aa24b45e4e58d3faa/checks/7550306979/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
doc tests for: /checkout/src/doc/rustc/src/platform-support/armv7-unknown-linux-uclibceabihf.md
doc tests for: /checkout/src/doc/rustc/src/platform-support/fuchsia.md


command did not execute successfully: "/checkout/obj/build/bootstrap/debug/rustdoc" "-Wrustdoc::invalid_codeblock_attributes" "-Dwarnings" "-Znormalize-docs" "-Z" "unstable-options" "--test" "/checkout/src/doc/rustc/src/platform-support/fuchsia.md" "--test-args" ""

stdout ----

running 5 tests
running 5 tests
test /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Testing (line 266) ... FAILED
test /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Create_a_package (line 133) ... FAILED
test /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Bulding_a_component (line 175) ... FAILED
test /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Create_a_package (line 109) ... FAILED
test /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs (line 89) ... ok
failures:


---- /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Testing (line 266) stdout ----
error: struct literal body without path
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:269:14
5 |       program: {
  |  ______________^
6 | |         runner: "elf",
6 | |         runner: "elf",
7 | |         binary: "bin/hello_fuchsia",
8 | |         args: ["it_works"],
  | |_____^
  |
  |
help: you might have forgotten to add the struct literal inside the block
  |
5 ~     program: { SomeStruct {
6 |         runner: "elf",
7 |         binary: "bin/hello_fuchsia",
8 |         args: ["it_works"],
  |


error: expected type, found `"syslog/client.shard.cml"`
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:268:16
  |
4 |     include: [ "syslog/client.shard.cml" ],
  |            -   ^^^^^^^^^^^^^^^^^^^^^^^^^ expected type
  |            |
  |            tried to parse a type due to this type ascription
  |
  = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`

error: struct literal body without path
error: struct literal body without path
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:267:1
3  | / {
3  | / {
4  | |     include: [ "syslog/client.shard.cml" ],
5  | |     program: {
6  | |         runner: "elf",
9  | |     },
10 | | }
   | |_^
   |
   |
help: you might have forgotten to add the struct literal inside the block
   |
3  ~ { SomeStruct {
4  |     include: [ "syslog/client.shard.cml" ],
9  |     },
10 ~ } }
   |


error: aborting due to 3 previous errors

Couldn't compile the test.
---- /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Create_a_package (line 133) stdout ----
error: expected expression, found `$`
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:134:19
  |
3 | bin/hello_fuchsia=${WORK_DIR}/bin/hello_fuchsia

error: aborting due to previous error

Couldn't compile the test.
Couldn't compile the test.
---- /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Bulding_a_component (line 175) stdout ----
error: struct literal body without path
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:178:14
5 |       program: {
  |  ______________^
6 | |         runner: "elf",
6 | |         runner: "elf",
7 | |         binary: "bin/hello_fuchsia",
  | |_____^
  |
  |
help: you might have forgotten to add the struct literal inside the block
  |
5 ~     program: { SomeStruct {
6 |         runner: "elf",
7 |         binary: "bin/hello_fuchsia",
  |


error: expected type, found `"syslog/client.shard.cml"`
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:177:16
  |
4 |     include: [ "syslog/client.shard.cml" ],
  |            -   ^^^^^^^^^^^^^^^^^^^^^^^^^ expected type
  |            |
  |            tried to parse a type due to this type ascription
  |
  = note: `#![feature(type_ascription)]` lets you annotate an expression with a type: `<expr>: <type>`

error: struct literal body without path
error: struct literal body without path
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:176:1
3 | / {
3 | / {
4 | |     include: [ "syslog/client.shard.cml" ],
5 | |     program: {
6 | |         runner: "elf",
7 | |         binary: "bin/hello_fuchsia",
9 | | }
  | |_^
  |
  |
help: you might have forgotten to add the struct literal inside the block
  |
3 ~ { SomeStruct {
4 |     include: [ "syslog/client.shard.cml" ],
8 |     },
9 ~ } }
  |


error: aborting due to 3 previous errors

Couldn't compile the test.
---- /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Create_a_package (line 109) stdout ----
error: unknown start of token: \u{2523}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:113:1
6 | ┣━ bin
  | ^

error: unknown start of token: \u{2501}
error: unknown start of token: \u{2501}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:113:2
6 | ┣━ bin
  |  ^
  |
  |
help: Unicode character '━' (Box Drawings Heavy Horizontal) looks like '-' (Minus/Hyphen), but it is not
6 | ┣- bin
  |  ~

error: unknown start of token: \u{2503}
error: unknown start of token: \u{2503}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:114:1
7 | ┃  ┗━ hello_fuchsia
  | ^

error: unknown start of token: \u{2517}
error: unknown start of token: \u{2517}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:114:4
7 | ┃  ┗━ hello_fuchsia
  |    ^

error: unknown start of token: \u{2501}
error: unknown start of token: \u{2501}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:114:5
7 | ┃  ┗━ hello_fuchsia
  |     ^
  |
  |
help: Unicode character '━' (Box Drawings Heavy Horizontal) looks like '-' (Minus/Hyphen), but it is not
7 | ┃  ┗- hello_fuchsia
  |     ~

error: unknown start of token: \u{2523}
error: unknown start of token: \u{2523}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:115:1
8 | ┣━ meta
  | ^

error: unknown start of token: \u{2501}
error: unknown start of token: \u{2501}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:115:2
8 | ┣━ meta
  |  ^
  |
  |
help: Unicode character '━' (Box Drawings Heavy Horizontal) looks like '-' (Minus/Hyphen), but it is not
8 | ┣- meta
  |  ~

error: unknown start of token: \u{2503}
error: unknown start of token: \u{2503}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:116:1
9 | ┃  ┣━ package
  | ^

error: unknown start of token: \u{2523}
error: unknown start of token: \u{2523}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:116:4
9 | ┃  ┣━ package
  |    ^

error: unknown start of token: \u{2501}
error: unknown start of token: \u{2501}
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:116:5
9 | ┃  ┣━ package
  |     ^
  |
  |
help: Unicode character '━' (Box Drawings Heavy Horizontal) looks like '-' (Minus/Hyphen), but it is not
9 | ┃  ┣- package
  |     ~

error: unknown start of token: \u{2503}
error: unknown start of token: \u{2503}
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:117:1
10 | ┃  ┗━ hello_fuchsia.cm
   | ^

error: unknown start of token: \u{2517}
error: unknown start of token: \u{2517}
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:117:4
10 | ┃  ┗━ hello_fuchsia.cm
   |    ^

error: unknown start of token: \u{2501}
error: unknown start of token: \u{2501}
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:117:5
10 | ┃  ┗━ hello_fuchsia.cm
   |     ^
   |
   |
help: Unicode character '━' (Box Drawings Heavy Horizontal) looks like '-' (Minus/Hyphen), but it is not
10 | ┃  ┗- hello_fuchsia.cm
   |     ~

error: unknown start of token: \u{2517}
error: unknown start of token: \u{2517}
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:118:1
   |
11 | ┗━ hello_fuchsia.manifest

error: unknown start of token: \u{2501}
error: unknown start of token: \u{2501}
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:118:2
   |
11 | ┗━ hello_fuchsia.manifest
   |
   |
help: Unicode character '━' (Box Drawings Heavy Horizontal) looks like '-' (Minus/Hyphen), but it is not
   |
11 | ┗- hello_fuchsia.manifest
   |  ~

error: expected `;`, found `hello_fuchsia`
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:110:17
  |
3 | hello_fuchsia.rs
  |                 ^ help: add `;` here
4 | hello_fuchsia.cml
  | ------------- unexpected token

error: expected `;`, found `package`
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:111:18
  |
4 | hello_fuchsia.cml
  |                  ^ help: add `;` here
5 | package
  | ------- unexpected token

error[E0425]: cannot find value `hello_fuchsia` in this scope
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:110:1
  |
3 | hello_fuchsia.rs


error[E0425]: cannot find value `hello_fuchsia` in this scope
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:111:1
  |
4 | hello_fuchsia.cml

error[E0425]: cannot find value `package` in this scope
error[E0425]: cannot find value `package` in this scope
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:112:1
5 | package
  | ^^^^^^^ not found in this scope

error[E0425]: cannot find value `bin` in this scope
error[E0425]: cannot find value `bin` in this scope
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:113:4
6 | ┣━ bin
  |    ^^^ not found in this scope


error[E0425]: cannot find value `hello_fuchsia` in this scope
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:114:7
7 | ┃  ┗━ hello_fuchsia
  |       ^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `meta` in this scope
error[E0425]: cannot find value `meta` in this scope
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:115:4
8 | ┣━ meta
  |    ^^^^ not found in this scope

error[E0425]: cannot find value `package` in this scope
error[E0425]: cannot find value `package` in this scope
 --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:116:7
9 | ┃  ┣━ package
  |       ^^^^^^^ not found in this scope


error[E0425]: cannot find value `hello_fuchsia` in this scope
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:117:7
10 | ┃  ┗━ hello_fuchsia.cm
   |       ^^^^^^^^^^^^^ not found in this scope


error[E0425]: cannot find value `hello_fuchsia` in this scope
  --> /checkout/src/doc/rustc/src/platform-support/fuchsia.md:118:4
   |
11 | ┗━ hello_fuchsia.manifest

error: aborting due to 26 previous errors

For more information about this error, try `rustc --explain E0425`.
For more information about this error, try `rustc --explain E0425`.
Couldn't compile the test.

failures:
    /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Bulding_a_component (line 175)
    /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Create_a_package (line 109)
    /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Building_Rust_programs::Create_a_package (line 133)
    /checkout/src/doc/rustc/src/platform-support/fuchsia.md - _and_::Testing (line 266)
test result: FAILED. 1 passed; 4 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.17s


stderr ----
```

</details>

--- a/src/doc/rustc/src/SUMMARY.md
+++ b/src/doc/rustc/src/SUMMARY.md
@@ -21,6 +21,7 @@
     - [armv6k-nintendo-3ds](platform-support/armv6k-nintendo-3ds.md)
     - [armv7-unknown-linux-uclibceabi](platform-support/armv7-unknown-linux-uclibceabi.md)
     - [armv7-unknown-linux-uclibceabihf](platform-support/armv7-unknown-linux-uclibceabihf.md)
+    - [\*-fuchsia](platform-support/fuchsia.md)
     - [\*-kmc-solid_\*](platform-support/kmc-solid.md)
     - [m68k-unknown-linux-gnu](platform-support/m68k-unknown-linux-gnu.md)
     - [mips64-openwrt-linux-musl](platform-support/mips64-openwrt-linux-musl.md)

--- a/src/doc/rustc/src/platform-support/fuchsia.md
+++ b/src/doc/rustc/src/platform-support/fuchsia.md
@@ -1,0 +1,295 @@
+# `aarch64-fuchsia` and `x86_64-fuchsia`
+
+**Tier: 2**
+
+[Fuchsia] is a modern open source operating system that's simple, secure,
+updatable, and performant.
+
+[Fuchsia]: https://fuchsia.dev/
+
+## Target maintainers
+
+The [Fuchsia team]:
+
+[Fuchsia team]: https://team-api.infra.rust-lang.org/v1/teams/fuchsia.json
+
+- Tyler Mandry ([@tmandry](https://github.com/tmandry))
+- Dan Johnson ([@computerdruid](https://github.com/computerdruid))
+- David Koloski ([@djkoloski](https://github.com/djkoloski))
+- Andrew Pollack ([@andrewpollack](https://github.com/andrewpollack))
+- Joseph Ryan ([@P1n3appl3](https://github.com/P1n3appl3))
+
+As the team evolves over time, the specific members listed here may differ from
+the members reported by the API. The API should be considered to be
+authoritative if this occurs. Instead of pinging individual members, use
+`@rustbot ping fuchsia` to contact the team on GitHub.
+
+## Requirements
+
+This target is cross-compiled from a host environment. Development may be done
+from the [source tree] or using the Fuchsia SDK.
+
+[source tree]: https://fuchsia.dev/fuchsia-src/get-started/learn/build
+
+Fuchsia targets support std and follow the `sysv64` calling convention on
+x86_64. Fuchsia binaries use the ELF file format.
+
+## Building the target
+
+Before building Rust for Fuchsia, you'll need a clang toolchain that supports
+Fuchsia as well. A recent version (14+) of clang should be sufficient to compile
+Rust for Fuchsia.
+
+You'll also need a recent copy of the [Fuchsia SDK], which provides the tools
+and binaries required to build and link programs for Fuchsia.
+
+[Fuchsia SDK]: https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core
+
+x86-64 and AArch64 Fuchsia targets can be enabled using the following
+configuration.
+
+In `config.toml`, add:
+
+```toml
+[build]
+target = ["<host_platform>", "aarch64-fuchsia", "x86_64-fuchsia"]
+
+[target.x86_64-fuchsia]
+llvm-libunwind = "in-tree"
+
+[target.aarch64-fuchsia]
+llvm-libunwind = "in-tree"
+```
+
+Additionally, the following environment variables must be configured (for
+example, using a script like `config-env.sh`):
+
+```sh
+# Configure this environment variable to be the path to the downloaded SDK
+export SDK_PATH="<SDK path goes here>"
+
+export CFLAGS_aarch64_fuchsia="--target=aarch64-fuchsia --sysroot=${SDK_PATH}/arch/arm64/sysroot -I${SDK_PATH}/pkg/fdio/include"
+export CXXFLAGS_aarch64_fuchsia="--target=aarch64-fuchsia --sysroot=${SDK_PATH}/arch/arm64/sysroot -I${SDK_PATH}/pkg/fdio/include"
+export LDFLAGS_aarch64_fuchsia="--target=aarch64-fuchsia --sysroot=${SDK_PATH}/arch/arm64/sysroot -L${SDK_PATH}/arch/arm64/lib"
+export CARGO_TARGET_AARCH64_FUCHSIA_RUSTFLAGS="-C link-arg=--sysroot=${SDK_PATH}/arch/arm64/sysroot -Lnative=${SDK_PATH}/arch/arm64/sysroot/lib -Lnative=${SDK_PATH}/arch/arm64/lib"
+export CFLAGS_x86_64_fuchsia="--target=x86_64-fuchsia --sysroot=${SDK_PATH}/arch/x64/sysroot -I${SDK_PATH}/pkg/fdio/include"
+export CXXFLAGS_x86_64_fuchsia="--target=x86_64-fuchsia --sysroot=${SDK_PATH}/arch/x64/sysroot -I${SDK_PATH}/pkg/fdio/include"
+export LDFLAGS_x86_64_fuchsia="--target=x86_64-fuchsia --sysroot=${SDK_PATH}/arch/x64/sysroot -L${SDK_PATH}/arch/x64/lib"
+export CARGO_TARGET_X86_64_FUCHSIA_RUSTFLAGS="-C link-arg=--sysroot=${SDK_PATH}/arch/x64/sysroot -Lnative=${SDK_PATH}/arch/x64/sysroot/lib -Lnative=${SDK_PATH}/arch/x64/lib"
+```
+
+These can be run together in a shell environment by executing
+`(source config-env.sh && ./x.py install)`.
+
+## Building Rust programs
+
+After compiling Rust binaries, you'll need to build a component, package it, and
+serve it to a Fuchsia device or emulator. All of this can be done using the
+Fuchsia SDK.
+
+As an example, we'll compile and run this simple program on a Fuchsia emulator:
+
+**`hello_fuchsia.rs`**
+```rust
+fn main() {
+    println!("Hello Fuchsia!");
+}
+
+#[test]
+fn it_works() {
+    assert_eq!(2 + 2, 4);
+}
+```
+
+Create a new file named `hello_fuchsia.rs` and fill out its contents with that
+code.
+
+### Create a package
+
+On Fuchsia, a package is the unit of distribution for software. We'll need to
+create a new package directory where we will place files like our finished
+binary and any data it may need. The working directory will have this layout:
+
+```txt
+hello_fuchsia.rs
+hello_fuchsia.cml
+package
+┣━ bin
+┃  ┗━ hello_fuchsia
+┣━ meta
+┃  ┣━ package
+┃  ┗━ hello_fuchsia.cm
+┗━ hello_fuchsia.manifest
+```
+
+Make the `package`, `package/bin`, and `package/meta` directories and create the
+following files inside:
+
+**`package/meta/package`**
+```json
+{"name":"hello_fuchsia","version":0}
+```
+
+The `package` file describes our package's name and version number. Every
+package must contain one.
+
+**`package/hello_fuchsia.manifest`**
+```txt
+bin/hello_fuchsia=package/bin/hello_fuchsia
+lib/ld.so.1=<SDK_PATH>/arch/x64/sysroot/dist/lib/ld.so.1
+lib/libfdio.so=<SDK_PATH>/arch/x64/dist/libfdio.so
+meta/package=package/meta/package
+meta/hello_fuchsia.cm=package/meta/hello_fuchsia.cm
+```
+
+*Note: Relative manifest paths are resolved starting from the working directory
+of `pm`. Make sure to fill out `<SDK_PATH>` with the path to the downloaded
+SDK.*
+
+The `.manifest` file will be used to describe the contents of the package by
+relating their location when installed to their location on the file system. You
+can use this to make a package pull files from other places, but for this
+example we'll just be placing everything in the `package` directory.
+
+### Compiling a binary
+
+Using your freshly compiled `rustc`, you can compile a binary for Fuchsia using
+the following options:
+
+* `--target x86_64-fuchsia`/`--target aarch64-fuchsia`: Targets the Fuchsia
+  platform of your choice
+* `-Lnative ${SDK_PATH}/arch/${ARCH}/lib`: Link against Fuchsia libraries from
+  the SDK
+* `-Lnative ${SDK_PATH}/arch/${ARCH}/sysroot/lib`: Link against Fuchsia kernel
+  libraries from the SDK
+
+Putting it all together:
+
+```sh
+# Configure these for the Fuchsia target of your choice
+TARGET_ARCH="<x86_64-fuchsia|aarch64-fuchsia>"
+ARCH="<x64|aarch64>"
+
+rustc --target ${TARGET_ARCH} -Lnative=${SDK_PATH}/arch/${ARCH}/lib -Lnative=${SDK_PATH}/arch/${ARCH}/sysroot/lib -o package/bin/hello_fuchsia hello_fuchsia.rs
+```
+
+### Bulding a component
+
+On Fuchsia, components require a component manifest written in Fuchia's markup
+language called CML. The Fuchsia devsite contains an [overview of CML] and a
+[reference for the file format]. Here's a basic one that can run our single binary:
+
+[overview of CML]: https://fuchsia.dev/fuchsia-src/concepts/components/v2/component_manifests
+[reference for the file format]: https://fuchsia.dev/reference/cml
+
+**`hello_fuchsia.cml`**
+```txt
+{
+    include: [ "syslog/client.shard.cml" ],
+    program: {
+        runner: "elf",
+        binary: "bin/hello_fuchsia",
+    },
+}
+```
+
+Now we can compile that CML into a component manifest:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/cmc compile hello_fuchsia.cml --includepath ${SDK_PATH}/pkg -o package/meta/hello_fuchsia.cm
+```
+
+`--includepath` tells the compiler where to look for `include`s from our CML.
+In our case, we're only using `syslog/client.shard.cml`.
+
+### Building and publishing a package
+
+Next, we'll build our package as defined by our manifest:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/pm -o hello_fuchsia -m package/hello_fuchsia.manifest build -output-package-manifest hello_fuchsia_manifest
+```
+
+This will produce `hello_fuchsia_manifest` which is a package manifest we can
+publish directly to a repository. We can set up that repository with:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/pm newrepo -repo repo
+```
+
+And then publish our new package to that repository with:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/pm publish -repo repo -lp -f <(echo "hello_fuchsia_manifest")
+```
+
+Then we can add it to `ffx`'s package server as `hello-fuchsia` using:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/ffx repository add-from-pm repo -r hello-fuchsia
+```
+
+### Starting the emulator
+
+Start a Fuchsia emulator in a new terminal using:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/ffx product-bundle get workstation_eng.qemu-${ARCH}
+${SDK_PATH}/tools/${ARCH}/ffx emu start workstation_eng.qemu-${ARCH} --headless
+```
+
+Then, once the emulator has been started:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/ffx target repository register
+```
+
+And watch the logs from the emulator in a separate terminal:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/ffx log --since now
+```
+
+Finally, run the component:
+
+```sh
+${SDK_PATH}/tools/${ARCH}/ffx component run fuchsia-pkg://hello-fuchsia/hello_fuchsia#meta/hello_fuchsia.cm
+```
+
+On reruns of the component, the `--recreate` argument may also need to be
+passed.
+
+## Testing
+
+### Running unit tests
+
+Tests can be run in the same way as a regular binary, simply by passing `--test`
+to the `rustc` invocation and then repackaging and rerunning. The test harness
+will run the applicable unit tests.
+
+Often when testing, you may want to pass additional command line arguments to
+your binary. Additional arguments can be set in the component manifest:
+
+**`hello_fuchsia.cml`**
+```txt
+{
+    include: [ "syslog/client.shard.cml" ],
+    program: {
+        runner: "elf",
+        binary: "bin/hello_fuchsia",
+        args: ["it_works"],
+    },
+}
+```
+
+This will pass the argument `it_works` to the binary, filtering the tests to
+only those tests that match the pattern. There are many more configuration
+options available in CML including environment variables. More documentation is
+available on the [Fuchsia devsite](https://fuchsia.dev/reference/cml).
+
+### Running the compiler test suite
+
+Running the Rust test suite on Fuchsia is [not currently supported], but work is
+underway to enable it.
+
+[not currently supported]: https://fxbug.dev/105393

